device/core/src/twin.ts

```diff
@@ -26,183 +33,106 @@ import { Client } from './client';
  *
  */
 export class Twin extends EventEmitter {
-  static timeout: number = 120000;
   static errorEvent: string = 'error';
-  static subscribedEvent: string = 'subscribed';
-  static responseEvent: string = 'response';
-  static postEvent: string = 'post';
   static desiredPath: string = 'properties.desired';
 
   /**
    * The desired and reported properties dictionnaries (respectively in `properties.desired` and `properties.reported`).
    */
-  properties: any; // TODO: need type
-
-  private _rid: number;
-  private _client: Client;
-  private _receiver: any; // TODO: need type
+  properties: TwinProperties;
+  desiredPropertiesUpdatesEnabled: boolean;
+  private _transport: Client.Transport;
+  private _retryPolicy: RetryPolicy;
+  private _maxOperationTimeout: number;
 
   /**
    * The constructor should not be used directly and instead the SDK user should use the {@link Client#getTwin} method to obtain a valid `Twin` object.
    * @constructor
    * @private
    * @param client The device client to use in order to communicate with the Azure IoT hub.
```

Comment:
>@param client The device client to use in order to communicate with the Azure IoT hub. [](start = 5, length = 86)

please update jsdoc.  I'm also really intrigued, and maybe even impressed, that you don't have a backpointer into the client. (reserving judgement until I read the code :))

Reply:
woops, good catch

---
In reply to: [171975982](https://github.com/pierreca/azure-iot-sdk-node/pull/2#discussion_r171975982) [](ancestors = 171975982)